lib/features/keyboard/DiagramKeyBindings.js

```diff
@@ -0,0 +1,134 @@
+import {
+  isCmd,
+  isShift,
+  MoveCanvasFactory
+} from './helpers';
+
+var zKeys = [ 'Z', 'z' ];
+
+var copyKeys = [ 'C', 'c' ];
+var pasteKeys = [ 'V', 'v' ];
+
+var undoKeys = [ 'Z', 'z' ];
+var redoKeys = zKeys.concat('Y', 'y');
+
+var removeSelectionKeys = [ 'Delete' ];
+
+var zoomDefaultKeys = [ '0' ];
+var zoomInKeys = [ '+', 'Add', '=' ];
+var zoomOutKeys = [ '-', 'Subtract' ];
+
+var moveCanvasLeftKeys = [ 'ArrowLeft', 'Left' ];
+var moveCanvasUpKeys = [ 'ArrowUp', 'Up' ];
+var moveCanvasRightKeys = [ 'ArrowRight', 'Right' ];
+var moveCanvasDownKeys = [ 'ArrowDown', 'Down' ];
+
+var DEFAULT_PRIORITY = 1000;
+
+/**
+ * Adds default KeyboardEvent listeners
+ *
+ * @param {Keyboard} keyboard
+ * @param {EditorActions} editorActions
+ */
+export default function DiagramKeyBindings(keyboard, editorActions) {
+
+  var moveCanvasLeft = MoveCanvasFactory('left', keyboard._config, editorActions);
```

Comment:
Why don't we plug in move canvas keyboard shortcuts via the MoveCanvas feature?

This would potentially allow us to easier override it (i.e. to restore the old behavior).

Reply:
It is actually done using the `MoveCanvas` feature, see: https://github.com/bpmn-io/diagram-js/pull/289/files#diff-ab467edcc671fb2f7e5cb7a91b06437d
The factory function was added to keep code DRY as the only difference between the functions responsible for the movement is the `direction` property passed to `MoveCanvas` feature. Alternatively, we could move the factory function to `DiagramKeyBindings.js` as a helper at the end of the file.

However, this does not seem to make it more difficult to override the listeners, because the way to do that is always the same: add listeners for the same keys but with higher priority.

Please let me know what you think.

Reply:
I'd also go for a `MoveCanvas` feature instead of a factory function.